### PR TITLE
Add test cases for the work item's view details component

### PR DESCRIPTION
**Pull request by nimishamukherjee**
Reference #35  - Add test cases for viewing a work item's details.

PR https://github.com/almighty/almighty-ui/pull/143 needs to be merged before this PR as it fixes a bug.

### CODE REVIEW

**Comment by nimishamukherjee** [comment]
@naina-verma please review this PR

**Test case covered:**
* Page title should have the work item title
* Page should display work item title
* Save should be enabled if a valid work item title has been entered
* Save should be disabled if work item's title has only whitespace
* Save should be disabled if the work item's title is blank
* Navigate when Back is clicked
* Navigate when click save and save resolves

**Files:**
* Component file : https://github.com/nimishamukherjee/almighty-ui/blob/alm-ui-35unittest4viewdetail/src/app/work-item/work-item-detail/work-item-detail.component.ts
* Test case file : https://github.com/nimishamukherjee/almighty-ui/blob/alm-ui-35unittest4viewdetail/src/app/work-item/work-item-detail/work-item-detail.component.spec.ts

work-item-detail.component.spec.ts has test cases written for work-item-detail.component.ts

**Comment by naina-verma** [comment]
@nimishamukherjee ,
You can add test for ID other wise this looks good to me !! 
:+1:

**Comment by naina-verma** [comment]
@nimishamukherjee ,Test cases for edit work items  

- Id should be ready only
- Test title Edit works 
- Test description edit works 
- Test title with Space / Blank throws validation (Mandatory field)
- Test Save updates the changes 
- Test user is able to select the Type
- Test user is able to modify Creator
- Test user is able to modify Assignee
- Test user is able to update State

**Comment by ldimaggi** [comment]
Just wondering - for a text input field, do we have a standard set of negative tests - things like string too long, spaces in strings, non-ASCII characters, multi-byte characters?  Thx!

File: src/app/work-item/work-item-detail/work-item-detail.component.html [hunk 1]
```diff
@@ -10,10 +10,12 @@
         {{workItem.id}}
       </div>
     </div>
-    <div class="form-group">
+    <div class="form-group" [ngClass]="{'has-error': !workItem.fields['system.title']}">
```
Comment (joshuawilson) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: It looks like this is fix from PR (#119). If you need this fix in here for the PR to work then the commits need to be merged. If you don't need it then it should be removed or at least a PR merge order should be listed. (something like please merge 119 before this one).

**Comment by ppitonak** [comment]
Jenkins job failed, any idea why? 

```
+ docker run --detach=true --name=almighty-ui-builder -e API_URL=http://demo.api.almighty.io/api/ -t -v /root/payload/dist:/dist:Z almighty-ui-builder
Unable to find image 'almighty-ui-builder:latest' locally
Trying to pull repository docker.io/library/almighty-ui-builder ... 
Pulling repository docker.io/library/almighty-ui-builder
Error: image library/almighty-ui-builder not found
docker: Error: image library/almighty-ui-builder not found.
See '/usr/bin/docker-current run --help'.
+ docker exec almighty-ui-builder npm install
Error response from daemon: No such container: almighty-ui-builder
+ docker exec almighty-ui-builder ./run_unit_tests.sh
Error response from daemon: No such container: almighty-ui-builder
+ '[' 1 -eq 0 ']'
+ echo 'CICO: unit tests FAIL'
CICO: unit tests FAIL
+ exit 1
```

**Comment by michaelkleinhenz** [comment]
@ppitonak which build was that? I fixed a problem today that caused this error. Basically the keyserver timed out resulting in a failed builder build. Should be fixed in master.

**Comment by nimishamukherjee** [comment]
PR https://github.com/almighty/almighty-ui/pull/143 needs to be merged before this PR as it fixes a bug.

**Comment by ppitonak** [comment]
@michaelkleinhenz it was this build https://ci.centos.org/job/devtools-almighty-ui/125/console
Latest build for this PR fails on failed test so the issue is solved.

**Comment by joshuawilson** [comment]
[test]

**Comment by joshuawilson** [comment]
@nimishamukherjee now that I merged PR #143 you need to rebase and merge the conflicting file.

**Comment by nimishamukherjee** [comment]
@joshuawilson I have rebased. The unit tests pass in the build process but the build fails.
Please check the console - https://ci.centos.org/job/devtools-almighty-ui/142/console

File: src/app/work-item/work-item-detail/work-item-detail.component.ts [hunk 2]
```diff
@@ -1,15 +1,14 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
 import { ActivatedRoute, Params } from '@angular/router';
-import { Location }               from '@angular/common';
```
Comment (joshuawilson) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: It appears that this file did not get rebased but instead reset to before the angular 2 update.

**Pull request merged**